addon/scope.js

```diff
@@ -0,0 +1,60 @@
+class Scope {
```

Comment:
I apologize if I'm not following this but am I correct that this implementation won't allow us to use any of the custom helpers we've built or any helpers that the user has built? If so, that seems like it would be confusing to users.